source/process/documentation-UItext-guidelines.rst

```diff
@@ -0,0 +1,193 @@
+   - If you're describing what just happened. For example, it's okay to say *Your incoming webhook has been set up*.
+   - If the subject (the doer of an action) is the Mattermost application itself. For example, it's okay to say *The image was deleted* instead of *The server deleted the image*.
+- Use the past tense to say what's already happened, the present tense to describe a current state or condition, and the future tense to state something that is very definitely going to happen.
+- If you're asking the user to do or not do something, use imperatives (command phrases). For example, say *Do not change the Hostname* instead of saying *It is not recommended to change the Hostname*. Better still, explain what could go wrong if they do or don't do something. For example, say *Do not change the Hostname because if you do so <this bad thing will happen>.*
+
+Guidelines for messages
+------------------------
+
+Try to use complete sentences in your messages. If a message contains variables (tokens):
+
+- Do not use verbs or adjectives as variables.
+- Do not create plurals of variables by adding an *s*.
+- If the variable is a noun, use a qualifier after the variable. For example, say *The {channel_name} channel was created* instead of saying *The {channel_name} was created*.
+
+Messages can be of the following types: notification, confirmation, warning, and error. The following sections contain guidelines that are specific to each of these types.
+
+Notification messages
+~~~~~~~~~~~~~~~~~~~~~~
+
+A notification message is for informing somebody about something that happened. Such messages do not need any user input, and do not prevent a user from continuing to use Mattermost.
+
+- Use either a complete sentence or a sentence phrase. 
+- If using a complete sentence, end it with a period.
+- Examples:
+
+  - *Member added to channel*
+  - *The plug-in was installed.*
+
+Confirmation messages
+~~~~~~~~~~~~~~~~~~~~~~
+
+A confirmation message is for asking somebody to confirm whether the immediately preceding command should be proceeded with. The user cannot use Mattermost until a confirmation is given or denied.
+
+- Use complete sentences.
+- Include a question that has a Yes/No answer.
+- Examples:
+
+  - *Are you sure you want to delete this channel?*
+  - *A plug-in with this ID already exists. Would you like to overwrite it?*
+
+Warning messages
+~~~~~~~~~~~~~~~~
+
+A warning message is for alerting somebody about something that might go wrong. The user can continue using Mattermost unless the warning message needs an explicit user input.
+
+- Use complete sentences.
+- Explain what has happened or can happen, and what can go wrong as a consequence.
+- If the message contains a question, phrase it in such a manner so that it has a Yes/No answer (unless you have specific action buttons for the message).
```

Comment:
Should the message contain a question if the user doesn't have to take action (since it's a warning)?

Reply:
I don't think so.

Reply:
In that case should we update the phrase

> If the message contains a question, phrase it in such a manner so that it has a Yes/No answer

Seems that warnings should not contain a question, since there are no actions

Reply:
*facepalm* yes! forgot to update. Corrected.